lib/engine/game/g_18_oe/game.rb

```diff
@@ -231,6 +457,14 @@ def metropolis_tile?(tile)
              OE18 OE26 OE27 OE28 OE29 OE30 OE37 OE38 OE39 OE40 OE41].include?(tile.name.to_s)
         end
 
+        def must_buy_train?(entity)
+          # must buy the reserved 2+2, otherwise only majors must buy trains
+          # return unless entity.type == 'major'
+          return if depot.depot_trains[0]&.name != '2+2' && entity.type != 'major'
```

Comment:
The type attribute returns a label, not a string, so you need to be testing for `:major`.
Also, it's clearer to use `Array.first` to return the first item in an array.
```
          return if depot.depot_trains.first&.name != '2+2' && entity.type != :major
```

Reply:
Yeah, I'm still relatively new to Ruby so knowing when I need to use labels vs strings is tricky for me

Reply:
These label/string mismatches don't get picked up when running in hotseat mode – all the code is converted by Opal into Javascript, which only has strings. But when running the game in server mode they will break.
Try creating a game in server mode with a few dummy players. This will flush these errors out.